apps/dashboard/src/components/contract-components/fetchPublishedContractsFromDeploy.ts

```diff
@@ -17,11 +28,35 @@ export async function fetchPublishedContractsFromDeploy(options: {
     throw new Error("No IPFS URI found in bytecode");
   }
 
-  const publishURIs = await getPublishedUriFromCompilerUri({
+  let publishURIs = await getPublishedUriFromCompilerUri({
     contract: getContractPublisher(client),
     compilerMetadataUri: contractUri,
   });
 
+  // Try fetching using contract name from compiler metadata for zksolc variants
+  // TODO: ContractPublisher should handle multiple metadata uri for a published version
+  if (publishURIs.length === 0 && (await isZkSyncChain(contract.chain))) {
+    try {
+      const deployMetadata = await download({
+        uri: contractUri,
+        client,
+      }).then((res) => res.json() as unknown as ZkSolcMetadata);
+
+      const contractId = Object.values(
+        deployMetadata.source_metadata.settings.compilationTarget,
+      );
+
+      if (contractId[0]) {
+        const published = await fetchPublishedContract({
+          client,
+          contractId: contractId[0],
+          publisherAddress: THIRDWEB_DEPLOYER_ADDRESS,
+        });
+        publishURIs = [published.publishMetadataUri];
+      }
+    } catch {}
```

Comment:
do we want to log an error here?

also what happens if this fails, but the code continues? is it just gonna fail down the line?

Reply:
Yeah this is a fallback, but we don't need to throw here.

Reply:
It's something I added in case there's a contract-id we can search on thirdweb published contracts. Else, continue as before.

Reply:
got it